packages/@aws-cdk/aws-appsync/lib/schema.ts

```diff
@@ -1,219 +1,55 @@
-    if (this.mode !== SchemaMode.CODE) {
-      throw new Error(`Unable to add query. Schema definition mode must be ${SchemaMode.CODE}. Received: ${this.mode}`);
-    }
-    if (!this.query) {
-      this.query = new ObjectType('Query', { definition: {} });
-      this.addType(this.query);
-    };
-    this.query.addField({ fieldName, field });
-    return this.query;
+    return new AssetSchemaImpl(filePath);
   }
 
   /**
-   * Add a mutation field to the schema's Mutation. CDK will create an
-   * Object Type called 'Mutation'. For example,
-   *
-   * type Mutation {
-   *   fieldName: Field.returnType
-   * }
-   *
-   * @param fieldName the name of the Mutation
-   * @param field the resolvable field to for this Mutation
-   */
-  public addMutation(fieldName: string, field: ResolvableField): ObjectType {
-    if (this.mode !== SchemaMode.CODE) {
-      throw new Error(`Unable to add mutation. Schema definition mode must be ${SchemaMode.CODE}. Received: ${this.mode}`);
-    }
-    if (!this.mutation) {
-      this.mutation = new ObjectType('Mutation', { definition: {} });
-      this.addType(this.mutation);
-    };
-    this.mutation.addField({ fieldName, field });
-    return this.mutation;
-  }
-
-  /**
-   * Add a subscription field to the schema's Subscription. CDK will create an
-   * Object Type called 'Subscription'. For example,
-   *
-   * type Subscription {
-   *   fieldName: Field.returnType
-   * }
-   *
-   * @param fieldName the name of the Subscription
-   * @param field the resolvable field to for this Subscription
-   */
-  public addSubscription(fieldName: string, field: ResolvableField): ObjectType {
-    if (this.mode !== SchemaMode.CODE) {
-      throw new Error(`Unable to add subscription. Schema definition mode must be ${SchemaMode.CODE}. Received: ${this.mode}`);
-    }
-    if (!this.subscription) {
-      this.subscription = new ObjectType('Subscription', { definition: {} });
-      this.addType(this.subscription);
-    }
-    const directives = field.fieldOptions?.directives?.filter((directive) => directive.mutationFields);
-    if (directives && directives.length > 1) {
-      throw new Error(`Subscription fields must not have more than one @aws_subscribe directives. Received: ${directives.length}`);
-    }
-    this.subscription.addField({ fieldName, field });
-    return this.subscription;
-  }
-
-  /**
-   * Add type to the schema
-   *
-   * @param type the intermediate type to add to the schema
+   * Point GraphQL Schema to a S3 Location
    *
+   * @returns `Schema` with immutable schema defintion
+   * @param s3Location the s3 location of the schema
    */
-  public addType(type: IIntermediateType): IIntermediateType {
-    if (this.mode !== SchemaMode.CODE) {
-      throw new Error('API cannot add type because schema definition mode is not configured as CODE.');
-    }
-    this.types.push(type);
-    return type;
+  public static fromS3Location(s3Location: S3Location): Schema {
```

Comment:
Make these positional arguments instead, the interface doesn't buy us much here.